### PR TITLE
solved the issue mentioned in #174

### PR DESCRIPTION
**Pull request by AneshkaGoyal**
I have solved the issue opened as #174

### CODE REVIEW

**Comment by AneshkaGoyal** [comment]
> Thanks for your pull request. It looks like this may be your first contribution to a Google open source project (if not, look below for help). Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).
> 
> 📝 **Please visit https://cla.developers.google.com/ to sign.**
> 
> Once you've signed (or fixed any issues), please reply here with `@googlebot I signed it!` and we'll verify it.
> 
> #### What to do if you already signed the CLA
> ##### Individual signers
> * It's possible we don't have your GitHub username or you're using a different email address on your commit. Check [your existing CLA data](https://cla.developers.google.com/clas) and verify that your [email is set on your git commits](https://help.github.com/articles/setting-your-email-in-git/).
> 
> ##### Corporate signers
> * Your company has a Point of Contact who decides which employees are authorized to participate. Ask your POC to be added to the group of authorized contributors. If you don't know who your Point of Contact is, direct the Google project maintainer to [go/cla#troubleshoot](http://go/cla#troubleshoot) ([Public version](https://opensource.google/docs/cla/#troubleshoot)).
> * The email used to register you as an authorized contributor must be the email used for the Git commit. Check [your existing CLA data](https://cla.developers.google.com/clas) and verify that your [email is set on your git commits](https://help.github.com/articles/setting-your-email-in-git/).
> * The email used to register you as an authorized contributor must also be [attached to your GitHub account](https://github.com/settings/emails).
> 
> ℹ️ **Googlers: [Go here](https://goto.google.com/prinfo/https%3A%2F%2Fgithub.com%2Fflutter%2Fudacity-course%2Fpull%2F175) for more info**.

@googlebot I signed it!

**Comment by AneshkaGoyal** [comment]
Hey 
Please have a look at the pull request raised @ianloic @gmoothart @sethladd

**Comment by AneshkaGoyal** [comment]
@googlebot please review pull request

**Comment by maryx** [comment]
Hi @AneshkaGoyal! Thank you for your patience. Can you please run dartfmt on the file?

File: course/12_error/solution_12_error/lib/category_route.dart [hunk 1]
```diff
@@ -193,7 +193,10 @@ class _CategoryRouteState extends State<CategoryRoute> {
         children: _categories.map((Category c) {
           return CategoryTile(
             category: c,
-            onTap: _onCategoryTap,
+            onTap:
+            c.name == apiCategory['name'] && c.units.isEmpty
```
Comment (domesticmouse) [inline course/12_error/solution_12_error/lib/category_route.dart:197]: I suspect this logic should be integrated into `_onCategoryTap`

**Review by domesticmouse — CHANGES_REQUESTED** [review]
Splitting logic between the `onTap` handler and the callback is kinda messy. Please integrate.

**Pull request merged**